src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -132,18 +132,17 @@ public CommonCompiler(CommandLineParser parser, string responseFile, string[] ar
         internal abstract Type Type { get; }
 
         /// <summary>
-        /// The assembly file version of this compiler, used in logo and /version output.
+        /// The version of this compiler with commit hash, used in logo and /version output.
         /// </summary>
-        internal virtual string GetAssemblyFileVersion()
+        internal string GetCompilerVersion()
         {
-            Assembly assembly = Type.GetTypeInfo().Assembly;
-            return GetAssemblyFileVersion(assembly);
+            return GetProductVersion(Type);
         }
 
-        internal static string GetAssemblyFileVersion(Assembly assembly)
+        internal static string GetProductVersion(Type type)
```

Comment:
>Type type [](start = 49, length = 9)

Consider changing the parameter type to `Assembly`.

Reply:
I didn't understand the suggestion. Wouldn't that mean that callers now all need to do `typeof(XYZ).Assembly` instead of just `typeof(XYZ)`?

---
In reply to: [282618373](https://github.com/dotnet/roslyn/pull/35543#discussion_r282618373) [](ancestors = 282618373)

Reply:
Yes. It makes it clear the string depends on the assembly only, not the type.